src/Sylius/Component/Core/Factory/ChannelFactory.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Factory;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+class ChannelFactory implements ChannelFactoryInterface
+{
+    /** @var FactoryInterface */
+    private $decoratedFactory;
+
+    /** @var string */
+    private $defaultCalculationStrategy;
+
+    public function __construct(FactoryInterface $decoratedFactory, string $defaultCalculationStrategy)
```

Comment:
IMHO current ChannelFactory should be injected
```suggestion
    public function __construct(Sylius\Component\Channel\Factory\ ChannelFactoryInterface $decoratedFactory, string $defaultCalculationStrategy)
```

Reply:
I try but i got :

>   [TypeError]                                                                  
  Argument 1 passed to Sylius\Component\Core\Factory\ChannelFactory::__constr  
  uct() must implement interface Sylius\Component\Core\Factory\ChannelFactory  
  Interface, instance of Sylius\Component\Resource\Factory\Factory given, cal  
  led in /home/travis/build/Sylius/Sylius/var/cache/test_cached/ContainerJNMx  
  Xga/getChannelFactoryInterfaceService.php on line 12  

:disappointed: